src/core/Core.cpp

```diff
@@ -849,29 +859,36 @@ void ManageCombatModeAnimations() {
 				}
 				
 				// Launch the arrow
-				
 				EERIE_LINKEDOBJ_UnLinkObjectFromObject(io->obj, arrowobj.get());
 				changeAnimation(io, 1, alist[ANIM_MISSILE_STRIKE]);
 				SendIOScriptEvent(nullptr, io, SM_STRIKE, "bow");
 				StrikeAimtime();
 				player.m_strikeAimRatio = player.m_bowAimRatio;
-				Entity * quiver = getInventoryItemWithLowestDurability("arrows", 1.f);
+
 				float poisonous = 0.f;
+				Entity * arrow = getInventoryItemWithLowestDurability("arrow", 1.f);
+				if (arrow) {
+					arrow->destroyOne();
+
+					poisonous = arrow->poisonous;
+				} else {
+					Entity * quiver = getInventoryItemWithLowestDurability("arrows", 1.f);
 				
-				if(quiver) {
-					poisonous = quiver->poisonous;
-					if(quiver->poisonous_count > 0) {
-						quiver->poisonous_count--;
+					if(quiver) {
```

Comment:
also no comparison to nullptr, either dont do it at all, or do it everywhere, consistency is key :P

Reply:
you forgot to fix that one to

`if (quiver != nullptr)`